### ISSUE TITLE
Error message for bad openstudio enum value is not clear

### ISSUE THREAD

**Issue by macumber** [NREL/OpenStudio#1741]
Error message when accidentally typing:

```
"WaterHeater:Mixed".to_IddObjectType
```

instead of 

```
"WaterHeater:Mixed".to_IddObjectType 
```

is 

```
eval:1:in `initialize': Unknown Value (RuntimeError)
```

We should mention the string and the enum name in the message to better point users to what the issue is

**Comment by jmarrec** [comment]
You have twice the same command above. But yeah, the error message is still the same:

```
(main)> "WaterHeaterMixed".to_IddObjectType
RuntimeError: Unknown Value
from eval:1:in `initialize'
```

**Comment by jmarrec** [comment]
Fixed in develop3 via #3777
